### PR TITLE
MET-4785 Changed creation of temporary id for zip records

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java [hunk 1]
```diff
@@ -331,4 +332,9 @@ private RecordInfo handleDuplicated(String providerId, Step step, Record.RecordB
     saveErrorWhileHarvesting(recordToHarvest, providerId, step, new RuntimeException(recordErrorCreated.getMessage()));
     return null;
   }
+
+  private String createTemporaryIdFromPath(String extractedDirectory, Path pathToRelativize){
+    return !extractedDirectory.isBlank() ? Paths.get(extractedDirectory).relativize(pathToRelativize).toString() :
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java:337]: Suggestion to change the `if` to a positive expression to make it easy to read :wink: 
```
return extractedDirectory.isBlank() ? pathToRelativize.getFileName().toString() : Paths.get(extractedDirectory).relativize(pathToRelativize).toString();
            ;
```

File: src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java [hunk 2]
```diff
@@ -331,4 +332,9 @@ private RecordInfo handleDuplicated(String providerId, Step step, Record.RecordB
...
+            pathToRelativize.getFileName().toString();
+  }
 }
```
Comment (jeortizquan) [inline src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java:340]: I ran it locally and is asking for the `TestUtils` 
.../metis-sandbox/src/test/java/eu/europeana/metis/sandbox/common/TestUtils.java
to implement this methods the `TestHttpRecordIterator` class.
```
    @Override
    public String getExtractedDirectory() {
      return this.extractedDirectory != null ? this.extractedDirectory.toString() : "";
    }
```

**Review by jeortizquan — COMMENTED** [review]
We are good just are pending the unit test update.

File: src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java (hunk 1 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java:337]: Updated

File: src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java (hunk 2 above)
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java:340]: Oops! I don't know how I missed that, I'll fix it

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**